fendermint/vm/interpreter/src/fvm/genesis.rs

```diff
@@ -228,6 +230,26 @@ where
             )
             .context("failed to create reward actor")?;
 
+        // STAGE 1b: Then we initialize the in-repo actors.
+        //
+
+        // Initialize the chain metadata actor which handles saving metadata about the chain
+        // (e.g. block hashes) which we can query.
+        let empty_blockhashes_cid =
+            Array::<(), _>::new_with_bit_width(state.store(), BLOCKHASHES_AMT_BITWIDTH).flush()?;
+        state
+            .create_actor(
+                fendermint_actors::CHAINMETADATA_ACTOR_CODE_ID,
+                fendermint_actors::CHAINMETADATA_ACTOR_ID,
+                &fendermint_actor_chainmetadata::State {
+                    blockhashes: empty_blockhashes_cid,
+                    lookback_len: fendermint_actor_chainmetadata::DEFAULT_LOOKBACK_LEN,
```

Comment:
This could be moved into the genesis file, even if all we do is assign the default there and let them edit it manually.

Reply:
@aakoshh What about having something like a `custom_actors` field in the genesis file which allows us to more easily configure this (and future) actors more easily, something like:


```
"custom_actors": {
  "chainmetadata": {
     "lookback_len": 256,
     ...
  },
  ...
}
```

But maybe we can do that in a separate PR as this one is getting quite large and we can use the fixed value until then?

Reply:
Sure thing, a separate PR is always fine. 

In the future I imagine that we would want to make the genesis file itself generic so that users can inject some custom bits of handling, have their own actors initialised in the genesis process. I am not sure what's the most ergonomic way of doing that: to make it like a `Genesis<T: DeserializeOwned>` where a certain field is parsed from JSON into their type and given to their handler, or maybe we store stuff in a `BTreeMap<ActorName, Json>` and give that to them to parse, and one of the entires is our `chainmetadata` actor. 

For sure our actor _has_ to be in the genesis file because without it we can't implement the EVM actor completely. With the `Genesis<T>` approach it would have to live separately, with the map it can be one of many. 

Okay, I like that map-like approach in that it forces us to work out the code to support this use case and then it's easier to make it available for users as well.